packages/app/src/specs/SpecNameDisplay.vue

```diff
@@ -0,0 +1,17 @@
+<template>
+  <div
+    class="max-w-60 truncate overflow-hidden"
+    data-cy="spec-filename"
+  >
+    <span class="font-semibold text-gray-800">{{ props.specFileName }}</span><span class="text-gray-600">{{ props.specFileExtension }}</span>
+  </div>
+</template>
+
+<script lang="ts" setup>
+
+const props = defineProps<{
+  specFileName: string
+  specFileExtension: string
+}>()
+
+</script>
```

Comment:
The `SpecNameDisplay.vue` component is very similar to the filename logic in `SpecItem.vue` (lines 10-26). There is a bit of different functionality with Highlighter in `SpecItem.vue`, but in the areas where they have crossover, would there be benefit in sharing this logic?

Reply:
Good callout @rachelruderman. I started looking into consolidating these and ran into some performance issues - `SpecItem` uses the `HighlightedText` component to handle highlighting fuzzy search results so I had to use a `<component :is="..."` to dynamically build the inner content for the spec file name (since in my use case it's just pure text without highlighting), and this additional render step caused pretty painful delays in render on search. Maybe there's a better way to handle dynamic inner content here that I don't know about?